### PR TITLE
feat(deps): new step one of verified in person flow

### PR DESCRIPTION
**Pull request by bryce-mcmath**
New step 1 content of verified in person flow (alt person flow)

Includes the recommended patterns of preferring components over screens, letting screens only handle navigation mechanics, and modularizing group sections of code. Tests included, snapshot and UI. Here's what it looks like:

verified_in_person

### CODE REVIEW

**Comment by cvarjao** [comment]
@bryce-mcmath , is there a dev mode toggle to enabled/disable this new flow?

**Comment by bryce-mcmath** [comment]
> @bryce-mcmath , is there a dev mode toggle to enabled/disable this new flow?

@cvarjao yes, that was done in a previous PR. It only uses this flow if `store.developer.enableAltPersonFlow` is true, which is toggleable in developer mode and by default off

File: app/src/modules/unified/components/CardButton.tsx [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+  onPress: () => void
+  testIDKey: string
+  accessibilityLabel: string
+  actionText: string
+  description: string
+  imgSrc?: ImageSourcePropType
+  style?: ViewStyle
+}
+
+const CardButton: React.FC<CardButtonProps> = ({
```
Comment (jleach) [inline app/src/modules/unified/components/CardButton.tsx]: Is this specific to a "Card". Just thinking what a card is might not make sense to other devs unless it's generic like a CredentnialCard?

File: app/src/modules/unified/contents/VerificationStep1/Birthdate.tsx [hunk 2]
```diff
@@ -0,0 +1,82 @@
+import { Button, ButtonType, testIdWithKey, useStore, useTheme } from '@hyperledger/aries-bifold-core'
+import { useCallback, useState } from 'react'
+import { useTranslation } from 'react-i18next'
+import { StyleSheet, ScrollView, View, Text } from 'react-native'
+import DatePicker from 'react-native-date-picker'
+import { SafeAreaView } from 'react-native-safe-area-context'
+
+import { BCDispatchAction, BCState } from '../../../../store'
+
+type ContentProps = {
```
Comment (jleach) [inline app/src/modules/unified/contents/VerificationStep1/Birthdate.tsx]: BirthdateContentProps?

File: app/src/modules/unified/contents/VerificationStep1/Birthdate.tsx [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+  onComplete: () => void
+}
+
+const BirthdateContent: React.FC<ContentProps> = ({ onComplete }: ContentProps) => {
```
Comment (jleach) [inline app/src/modules/unified/contents/VerificationStep1/Birthdate.tsx]: A little ambigious in the name. Maybe BirthdateInputContent. Either way I'm fine with this as is - just say'n.

File: app/src/modules/unified/contents/VerificationStep1/Instructions.tsx [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+    },
+    controlsContainer: {
+      marginBottom: 20,
+      marginTop: 'auto',
+      marginHorizontal: 20,
+      position: 'relative',
+    },
+    image: {
+      width: width - pagePadding * 2,
+      height: (width - pagePadding * 2) * 0.67,
```
Comment (jleach) [inline app/src/modules/unified/contents/VerificationStep1/Instructions.tsx]: Is "0.67" a magic number?

File: app/src/modules/unified/contents/VerificationStep1/Instructions.tsx [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+      ...TextTheme.normal,
+      marginBottom: 8,
+    },
+  })
+
+  return (
+    <SafeAreaView style={styles.pageContainer} edges={['bottom', 'left', 'right']}>
+      <ScrollView contentContainerStyle={styles.scrollView}>
+        <Image
+          source={require('../../assets/img/highlight_serial_barcode.png')}
```
Comment (jleach) [inline app/src/modules/unified/contents/VerificationStep1/Instructions.tsx]: Would this work? Using require in-line seems a bit odd to me.

```
import highlightSerialBarcode from '../../assets/img/highlight_serial_barcode.png';

// ... in your component ...
<Image source={highlightSerialBarcode} /> 
```

File: app/src/modules/unified/contents/VerificationStep1/ManualSerial.tsx [hunk 6]
```diff
@@ -0,0 +1,129 @@
...
+    setSerial(text)
+  }, [])
+
+  const onContinuePressed = useCallback(() => {
+    if (serial.length < 1) {
+      setErrorState({
+        description: t('Unified.ManualSerial.EmptySerialError'),
+        visible: true,
+      })
+    } else if (serial.length > 15) {
```
Comment (jleach) [inline app/src/modules/unified/contents/VerificationStep1/ManualSerial.tsx]: Is 15 a magic number? Is it `maxSerialNumberLenght`?

File: app/src/modules/unified/contents/VerificationStep1/ManualSerial.tsx [hunk 7]
```diff
@@ -0,0 +1,129 @@
...
+      width: '100%',
+    },
+  })
+
+  const handleChangeText = useCallback((text: string) => {
+    setSerial(text)
+  }, [])
+
+  const onContinuePressed = useCallback(() => {
+    if (serial.length < 1) {
```
Comment (jleach) [inline app/src/modules/unified/contents/VerificationStep1/ManualSerial.tsx]: I'd probably use early returns here:

**Improved readability** The code becomes flatter and easier to follow, as you avoid nested else if blocks.
**Reduced cognitive load** You don't have to keep track of the conditions in the if and else if to understand what happens in the else block.
**Clearer intent** Early returns explicitly show that the function should exit if a condition is met, making the logic more explicit.

File: app/src/modules/unified/contents/VerificationStep1/ManualSerial.tsx [hunk 8]
```diff
@@ -0,0 +1,129 @@
...
+        <View style={styles.contentContainer}>
+          <Image
+            source={require('../../assets/img/highlight_serial_barcode.png')}
+            style={styles.image}
+            resizeMode={'contain'}
+          />
+          <LimitedTextInput
+            defaultValue={serial}
+            label={t('Unified.ManualSerial.InputLabel')}
+            limit={15}
```
Comment (jleach) [inline app/src/modules/unified/contents/VerificationStep1/ManualSerial.tsx]: Repeated from if/else above? Should become a const.

File: app/src/modules/unified/contents/VerificationStep1/Scan.tsx [hunk 9]
```diff
@@ -0,0 +1,33 @@
+import { useTheme } from '@hyperledger/aries-bifold-core'
+import { Text, StyleSheet, ScrollView } from 'react-native'
+import { SafeAreaView } from 'react-native-safe-area-context'
+
+type ContentProps = {
```
Comment (jleach) [inline app/src/modules/unified/contents/VerificationStep1/Scan.tsx]: ScanContentProps?

File: app/src/modules/unified/contents/VerificationSteps/Steps.tsx [hunk 10]
```diff
@@ -1,22 +1,27 @@
...
   title: string
   active: boolean
   complete: boolean
   content: JSX.Element
   onPress?: () => void
+  testIDKey: string
 }
 
-const VerificationSteps: React.FC = () => {
+type ContentProps = {
```
Comment (jleach) [inline app/src/modules/unified/contents/VerificationSteps/Steps.tsx]: Repeated type name from Scan.tsx? `VerificationStepsContentProps`?

File: app/src/modules/unified/navigators/VerifiedPersonStack.tsx [hunk 11]
```diff
@@ -0,0 +1,118 @@
+import {
+  ButtonLocation,
+  IconButton,
+  testIdWithKey,
+  useDefaultStackOptions,
+  useTheme,
+} from '@hyperledger/aries-bifold-core'
+import { createStackNavigator } from '@react-navigation/stack'
+import { useTranslation } from 'react-i18next'
+import VerificationStep1 from '../screens/VerificationStep1'
```
Comment (jleach) [inline app/src/modules/unified/navigators/VerifiedPersonStack.tsx]: I feel like these will be hard for developers to reason about. `DOBCollectionStep`?

**Review by jleach — CHANGES_REQUESTED** [review]

File: app/src/modules/unified/components/CardButton.tsx (hunk 1 above)
Comment (bryce-mcmath) [reply app/src/modules/unified/components/CardButton.tsx]: @jleach I looked around a fair bit and Card seems to be common name for this kind of component (eg. https://mui.com/material-ui/react-card/) but I get that it's could sound similar to a credential card in our context. How about `PressableTile` or `TileButton`. Would that be better? I can't think of anything else to call it that's not >= current level of confusion.

File: app/src/modules/unified/contents/VerificationStep1/Instructions.tsx (hunk 5 above)
Comment (bryce-mcmath) [reply app/src/modules/unified/contents/VerificationStep1/Instructions.tsx]: Inline require seems like the way to do according to the docs: https://reactnative.dev/docs/images

I get errors if I move the require to the top of the file or if I turn it into an import statement

File: app/src/modules/unified/components/CardButton.tsx (hunk 1 above)
Comment (bryce-mcmath) [reply app/src/modules/unified/components/CardButton.tsx]: Done

File: app/src/modules/unified/contents/VerificationStep1/Birthdate.tsx (hunk 2 above)
Comment (bryce-mcmath) [reply app/src/modules/unified/contents/VerificationStep1/Birthdate.tsx]: Done

File: app/src/modules/unified/contents/VerificationStep1/Birthdate.tsx (hunk 3 above)
Comment (bryce-mcmath) [reply app/src/modules/unified/contents/VerificationStep1/Birthdate.tsx]: It's for all the content related to the birthdate view, not just the input component. Should be fine for now at least

File: app/src/modules/unified/contents/VerificationStep1/Instructions.tsx (hunk 4 above)
Comment (bryce-mcmath) [reply app/src/modules/unified/contents/VerificationStep1/Instructions.tsx]: Done

File: app/src/modules/unified/contents/VerificationStep1/ManualSerial.tsx (hunk 7 above)
Comment (bryce-mcmath) [reply app/src/modules/unified/contents/VerificationStep1/ManualSerial.tsx]: Done

File: app/src/modules/unified/contents/VerificationStep1/ManualSerial.tsx (hunk 8 above)
Comment (bryce-mcmath) [reply app/src/modules/unified/contents/VerificationStep1/ManualSerial.tsx]: Done

File: app/src/modules/unified/contents/VerificationStep1/Scan.tsx (hunk 9 above)
Comment (bryce-mcmath) [reply app/src/modules/unified/contents/VerificationStep1/Scan.tsx]: Done

File: app/src/modules/unified/contents/VerificationSteps/Steps.tsx (hunk 10 above)
Comment (bryce-mcmath) [reply app/src/modules/unified/contents/VerificationSteps/Steps.tsx]: Done

File: app/src/modules/unified/navigators/VerifiedPersonStack.tsx (hunk 11 above)
Comment (bryce-mcmath) [reply app/src/modules/unified/navigators/VerifiedPersonStack.tsx]: Done

**Review by jleach — APPROVED** [review]

**Pull request merged**